### PR TITLE
[PD] Fix Pad uptoshape and add unit test

### LINKED ISSUE #16700: PartDesign: Simple Pad Up to shape fails with "Could not create face from sketch" error
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I can't get a simple Pad with Up to shape mode to work (with all faces or with their manual selection):

up to shape

Same if the plane is on the other side. Previously, there were no problems with cases like this so it seems to be a regression after #16030.

```
15:19:00  Pad001: Could not create face from sketch.
Intersecting sketch entities or multiple faces in a sketch are not allowed.
```

[Pad up to shape fails.FCStd.zip](https://github.com/user-attachments/files/17084183/Pad.up.to.shape.fails.FCStd.zip)

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 1.1.0dev.38794 (Git)
Build type: Release
Branch: main
Hash: a47800d9ccd7d7a7cdb1779d56de03d3c64cd76e
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: Polish/Poland (pl_PL)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
```

### Subproject(s) affected?

PartDesign

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by FlachyJoe**
Fix #16700

### CODE REVIEW

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 1]
```diff
@@ -4231,10 +4231,12 @@ TopoShape& TopoShape::makeElementPrismUntil(const TopoShape& _base,
...
-    gp_Pnt pCog(vCog.x, vCog.y, vCog.z);
-    checkLimits = ! Part::Tools::isConcave(TopoDS::Face(__uptoface.getShape()), pCog , direction);
+    if (checkLimits && __uptoface.shapeType(true) == TopAbs_FACE){
+        Base::Vector3d vCog;
+        profile.getCenterOfGravity(vCog);
+        gp_Pnt pCog(vCog.x, vCog.y, vCog.z);
+        checkLimits = ! Part::Tools::isConcave(TopoDS::Face(__uptoface.getShape()), pCog , direction);
+    }
 
     TopoShape _uptoface(__uptoface);
```
Comment (bgbsww) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:4241]: So I really don't love this because it is so difficult for the reader.  But it isn't part of the PR.

**Review by bgbsww — APPROVED** [review]
This PR looks solid to me.  I hate the pre-existing confusing names in the method, bet that's asking for a refactor and not a bug fix.

**Comment by chennes** [comment]
Thanks for the review, @bgbsww -- I agree that the `TopoShape _uptoface(__uptoface);` is opaque, and that would be a nice future refactor. I've merged the PR and cherry-picked to 1.0.

File: src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 1 above)
Comment (FlachyJoe) [reply src/Mod/Part/App/TopoShapeExpansion.cpp:4241]: I agree too and I'm not sure all these copies are really needed.

**Pull request merged**